common/src/main/java/org/corfudb/common/protocol/client/ChannelHandler.java

```diff
@@ -41,13 +43,14 @@
  */
 
 @Slf4j
+@NoArgsConstructor
 public abstract class ChannelHandler extends ResponseHandler {
 
     //TODO(Maithem): what if the consuming client is using a different protobuf lib version?
 
-    protected final InetSocketAddress remoteAddress;
+    protected InetSocketAddress remoteAddress;
 
-    protected final EventLoopGroup eventLoopGroup;
```

Comment:
Restore these finals.

Reply:
We can't make them final and have a `@NoArgsConstructor` at the same time. We need `@NoArgsConstructor` for the tests so that we can set only what is required. The current constructor which takes those arguments makes a connect() call in the end which uses those values and we don't want a server connection for the unit tests.